### PR TITLE
Connect react and redux

### PR DESCRIPTION
**Pull request by Abe1able**
### In this milestone of the project, I have

- connected react and redux.
- imported Redux Provider and my store in the main component of my React app.
- connected my app to the Redux store.
- made sure that list of books displays books from the Redux store.
- used useSelector to consume the state.

### CODE REVIEW

File: src/components/Books/CreateNewBook.js [hunk 1]
```diff
@@ -13,19 +33,27 @@ const CreateNewBook = () => {
           type="text"
           placeholder="Book title"
           className="grid-item"
+          name="title"
+          value={inputBook.title}
+          onChange={onChange}
+          required
         />
```
Comment (cynthiainga) [inline src/components/Books/CreateNewBook.js]: - [x] The form for adding the book should have two inputs, one for the `title` and another for the `author`. Please kindly fix that 🙃 🙃

File: src/components/Categories.js [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+  const handleClick = () => {
+    dispatch(checkStatus());
+  };
+  return (
+    <div>
+      <button type="button" onClick={handleClick}>
+        Check Status
+      </button>
+      <p>{categories}</p>
+    </div>
```
Comment (cynthiainga) [inline src/components/Categories.js]: - [x] Please kindly make sure your button works as expected, normally when clicking on the `Check Status` button it's supposed to return the `Under construction` text under the button. But this is not the case for you, when clicking on the button nothing happens, please kindly fix that 😉

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @Abe1able, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] No linters errors ✔️ 
- [ ] Descriptive PR ✔️ 
- [ ] Correct Gitflow ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇 👇 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Categories.js [hunk 3]
```diff
@@ -1,8 +1,21 @@
...
+    dispatch(checkStatus());
+  };
+
+  return (
+    <div>
+      <button type="button" onClick={statusChecker}>Check status</button>
+      <h1>{statusCheck }</h1>
+    </div>
+  );
+}
```
Comment (cwaku) [inline src/components/Categories.js]: - [x]  As required by the previous reviewer, kindly make sure your button works as expected, normally when clicking on the Check Status button it's supposed to return the Under construction text under the button. But this is not the case for you, when clicking on the button nothing happens, please kindly fix that to have your project approved.

**Review by cwaku — CHANGES_REQUESTED** [review]
Hi @Abe1able 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

 ## Highlights
 - [x] Good job overall following the template design 💯 
 - [x] Correct GitHub Flow 👍🏾 
 - [x] Good commit history 🥇 
 - [x] Good PR title/description 💯 
 - [x] All linters passed 🥇 

## Required Changes ♻️
It's ok to have bugs in your code. Please, see this code review as a chance to learn practical coding skills and new tips 🌟  that can improve your code quality 💯 and help you get a better job 💼.

I will guide 📔 you through fixing 🛠️ these bugs, follow my comments down below and make sure you don't miss any of them to get your project approved ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. The previous reviewer simply missed some of them, and addressing them will improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books/CreateNewBook.js (hunk 1 above)
Comment (Abe1able) [reply src/components/Books/CreateNewBook.js]: Fixed, thank you.

File: src/components/Categories.js (hunk 2 above)
Comment (Abe1able) [reply src/components/Categories.js]: Fixed, thank you.

File: src/components/Categories.js (hunk 3 above)
Comment (Abe1able) [reply src/components/Categories.js]: Fixed, thank you.

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @Abe1able        :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**